x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/operator/arithmetic/Arithmetics.java

```diff
@@ -33,20 +39,12 @@ static Number add(Number l, Number r) {
         return Integer.valueOf(Math.addExact(l.intValue(), r.intValue()));
     }
 
-    static ZonedDateTime add(ZonedDateTime l, Period r) {
-        if (l == null || r == null) {
-            return null;
-        }
-
-        return l.plus(r);
+    static Temporal add(Temporal l, Period r) {
```

Comment:
Using `Temporal` is too broad without validation is dangerous. We actually accept only `ZonedDateTime` and `OffsetTime`. Ideally we should check this up to the source however this creates too many methods - yet at some point we need to do the "ugly" part of checking the actual types and throwing an exception otherwise.